tools/scripts/clang_tidy_check.sh

```diff
@@ -65,7 +65,57 @@ if [[ -n "$noSpaceInSuppressions" ]]; then
     exit 1
 fi
 
+function scanWithFileList() {
+    FILE_WITH_SCAN_LIST=$1
+    FILE_TO_SCAN=$2
+
+    if ! test -f "$FILE_WITH_SCAN_LIST"
+    then
+        echo "Scan list file '${FILE_WITH_SCAN_LIST}' does not exist"
+        exit 1
+    fi
+
+    for LINE in $(cat $FILE_WITH_SCAN_LIST); do
+        # add files until the comment section starts
+        if [[ "$(echo $LINE | grep "#" | wc -l)" == "1" ]]; then
+            break
+        fi
+        FILE_LIST="${FILE_LIST} $LINE"
+    done
+
+    if [[ -n $FILE_TO_SCAN ]]
+    then
+        if ! test -f "$FILE_TO_SCAN"
+        then
+            echo "The file which should be scanned '${FILE_TO_SCAN}' does not exist"
+            exit 1
+        fi
+
+        if [[ $(find ${FILE_LIST} -type f | grep -E ${FILE_FILTER} | grep ${FILE_TO_SCAN} | wc -l) == "0" ]]
+        then
+            echo "Skipping file '${FILE_TO_SCAN}' since it is not part of '${FILE_WITH_SCAN_LIST}'"
+            exit 0
+        fi
+
+        echo "Scanning file: ${FILE_TO_SCAN}"
+        $CLANG_TIDY_CMD --warnings-as-errors=* -p build $FILE_TO_SCAN
+    else
+        if [[ -z $FILE_LIST ]]
+        then
+            echo "'${FILE_WITH_SCAN_LIST}' is empty skipping folder scan."
+            exit 0
+        fi
+        echo "Performing full scan of all folders in '${FILE_WITH_SCAN_LIST}'"
+        $CLANG_TIDY_CMD --warnings-as-errors=* -p build $(find ${FILE_LIST} -type f | grep -E ${FILE_FILTER})
```

Comment:
```suggestion
        $CLANG_TIDY_CMD --warnings-as-errors=* -p build "$(find "${FILE_LIST}" -type f | grep -E ${FILE_FILTER})"
```

Reply:
Really? Bash is weird. How does it know which quotes belongs together?

Reply:
... oh and no quotes around `${FILE_FILTER}`?